application/src/main/java/org/togetherjava/tjbot/features/jshell/JShellCommand.java

```diff
@@ -0,0 +1,298 @@
+package org.togetherjava.tjbot.features.jshell;
+
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.events.interaction.ModalInteractionEvent;
+import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
+import net.dv8tion.jda.api.interactions.InteractionHook;
+import net.dv8tion.jda.api.interactions.callbacks.IReplyCallback;
+import net.dv8tion.jda.api.interactions.commands.OptionMapping;
+import net.dv8tion.jda.api.interactions.commands.OptionType;
+import net.dv8tion.jda.api.interactions.commands.build.SubcommandData;
+import net.dv8tion.jda.api.interactions.components.Modal;
+import net.dv8tion.jda.api.interactions.components.text.TextInput;
+import net.dv8tion.jda.api.interactions.components.text.TextInputStyle;
+import net.dv8tion.jda.api.interactions.modals.ModalMapping;
+import net.dv8tion.jda.api.utils.FileUpload;
+
+import org.togetherjava.tjbot.features.CommandVisibility;
+import org.togetherjava.tjbot.features.SlashCommandAdapter;
+import org.togetherjava.tjbot.features.jshell.backend.JShellApi;
+import org.togetherjava.tjbot.features.jshell.render.Colors;
+import org.togetherjava.tjbot.features.utils.RequestFailedException;
+
+import javax.annotation.Nullable;
+
+import java.util.List;
+import java.util.Objects;
+
+public class JShellCommand extends SlashCommandAdapter {
+    private static final String JSHELL_TEXT_INPUT_ID = "jshell";
+    private static final String JSHELL_COMMAND = "jshell";
+    private static final String JSHELL_VERSION_SUBCOMMAND = "version";
+    private static final String JSHELL_EVAL_SUBCOMMAND = "eval";
+    private static final String JSHELL_SNIPPETS_SUBCOMMAND = "snippets";
+    private static final String JSHELL_CLOSE_SUBCOMMAND = "shutdown";
+    private static final String JSHELL_STARTUP_SCRIPT_SUBCOMMAND = "startup-script";
+    private static final String JSHELL_CODE_PARAMETER = "code";
+    private static final String JSHELL_STARTUP_SCRIPT_PARAMETER = "startup-script";
+    private static final String JSHELL_USER_PARAMETER = "user";
+    private static final String JSHELL_INCLUDE_STARTUP_SCRIPT_PARAMETER = "include-startup-script";
+
+    private static final int MIN_MESSAGE_INPUT_LENGTH = 0;
+    private static final int MAX_MESSAGE_INPUT_LENGTH = TextInput.MAX_VALUE_LENGTH;
+
+    private final JShellEval jshellEval;
+
+    /**
+     * Creates an instance of the command.
+     */
+    public JShellCommand(JShellEval jshellEval) {
+        super(JSHELL_COMMAND, "JShell as a command.", CommandVisibility.GUILD);
```

Comment:
bad description. users already know its a command when reading this text. provide explanation what jshell is instead

Reply:
Improved, tell me what you think.